### ISSUE TITLE
CMake: please unbreak ccache handling

### ISSUE THREAD

**Issue by LebedevRI** [darktable-org/darktable#13856]
1. There is no configuration toggle for it
2. It should not default to `ON`
3. By default it spams with it's cache into `~/.ccache/ccache`, that's not great

Just because there's ccache binary does not mean everyone immediately wants to use it for everything.
If they did, they'd muck with symlinks so it's automatically picked by default instead of the compiler.

**Comment by parafin** [comment]
I side with @LebedevRI that it should be optional and default to off. If user wants to use ccache for everything, he will install symlinks. ccache (at least before) doesn't guarantee unbroken builds, and it spams ~/.ccache directory, which may be totally unexpected for the user (he may even not know that ccache is installed on his system).

**Comment by LebedevRI** [comment]
> @LebedevRI , the ccache handling in the CMakeLists.txt is fairly standard and is not broken, so the title of the issue is a little bit misleading, and expresses more of a personal opinion than a technical issue.

Correct.

> I need more context to understand your requests.

I'm simply stating which enhancements need to be made, and asking if you want to address them, else i will have to.

**Comment by LebedevRI** [comment]
FWIW, i very much agree that in general, compiler caching has it's benefits,
i just don't believe in forcibly inflicting good. There is a pretty narrow use-case
for ccache - when you are otherwise throughput-limited. E.g. i build darktable,
at most, once a week, so the cache is effectively never going to be used,
yet it's there and occupying space without my explicit permission.

There's also the question of discoverability - i don't see any messages whatsoever
in the cmake output about ccache being looked for, or used.
There's also the question of preference - why ccache? What if i want sccache?

TLDR: please either address these issues or revert the change.

**Comment by edgomez** [comment]
Well now that CMake requirement is 3.18 (since 2023-01-07), we can let the users specify CMAKE(C|CX)_COMPILER_LAUNCHER variables at configuration time.

I proposed a change to build.sh to let users do so easily.

Let me know if that's good for you all.

**Comment by LebedevRI** [comment]
https://cmake.org/cmake/help/v3.4/variable/CMAKE_LANG_COMPILER_LAUNCHER.html#variable:CMAKE_%3CLANG%3E_COMPILER_LAUNCHER suggests both of them appeared at the same time.
But i digress.
Thank you.

**Comment by edgomez** [comment]
@LebedevRI : indeed, i was mistaken, tried to find where I've got that 3.17 minimum version, and could only find the pages mentioning 3.4.

This means the initial patch could have been avoided alltogether, and really emphasizes the current choice not to meddle with the property directly is the right approach. But I ignored it existed. I need to revisit some profesional CMakeLists.txt next week :-)

**Comment by ptilopteri** [comment]
the inclusion of sccache option breaks obs building for Tumbleweed since darktable-4.3.0~git995.a468c256

error: line 114: Unknown tag: BuildFlags: ccachetype:sccache
